src/Services/Qdm/Services/ImmunizationAdministeredService.php

```diff
@@ -49,12 +49,19 @@ public function makeQdmModel(array $record)
             }
         }
 
-        $model->addCode(
-            new Code([
-                'code' => $record['cvx_code'],
-                'system' => $this->getSystemForCodeType('CVX')
-            ])
-        );
+        if (strpos($record['cvx_code'], ':') !== false) {
+            // Use the make code that blanks out the OID portion, or creates a code with code type
+            $model->addCode($this->makeQdmCode($record['cvx_code']));
+        } else {
+            // We make the code manually here, not using makeQdmCode because we need to set the code system
+            $model->addCode(
+                new Code([
+                    'code' => $record['cvx_code'],
+                    'system' => $this->getSystemForCodeType('CVX')
+                ])
+            );
+        }
```

Comment:
@kchapple This PR didn't get merged for some reason. This change seemed important and is not in current master.

Should it be?

Reply:
@sjpadgett I think I fixed it again during my last push

```
        if (str_starts_with($record['cvx_code'], 'OID:')) {
            // Sometimes codes are nulled out and sdc:valueSet is in the code, like this:
            // <code nullFlavor="NA" sdtc:valueSet="2.16.840.1.113883.3.526.3.1254"/>
            $model->addCode(
                new Code([
                    'code' => str_replace('OID:', '', $record['cvx_code']),
                    'system' => null
                ])
            );
        } else {
            $model->addCode(
                new Code([
                    'code' => $record['cvx_code'],
                    'system' => $this->getSystemForCodeType('CVX')
                ])
            );
        }
```

Reply:
This can be ignored and closed I think.

Reply:
> str_starts_with($record['cvx_code'], 'OID:')

Still though, this can be an issue if `$record['cvx_code']` is empty because for god only knows why, PHP parse will eval `str_starts_with` true for empty strings!
I'll fix that in my next PR. Thanks for getting back quickly.